packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -1,260 +0,0 @@
-
-var view, lookup, owner, TemplateTests, originalViewKeyword;
-
-QUnit.module('ember-htmlbars: {{#if}} and {{#unless}} helpers', {
-  setup() {
-    originalViewKeyword = registerKeyword('view',  viewKeyword);
-
-    Ember.lookup = lookup = {};
-    lookup.TemplateTests = TemplateTests = Namespace.create();
-    owner = buildOwner();
-    owner.registerOptionsForType('template', { instantiate: false });
-    owner.registerOptionsForType('view', { singleton: false });
-    owner.registerOptionsForType('component', { singleton: false });
-    owner.register('view:toplevel', EmberView.extend());
-    owner.register('component-lookup:main', ComponentLookup);
-  },
-
-  teardown() {
-    runDestroy(owner);
-    runDestroy(view);
-    owner = view = null;
-
-    Ember.lookup = lookup = originalLookup;
-    TemplateTests = null;
-
-    resetKeyword('view', originalViewKeyword);
-  }
-});
-
-QUnit.test('properties within an if statement should not fail on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.value}}{{view.value}}{{/if}}'),
-    value: null
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('value', 'test');
-  });
-
-  equal(view.$().text(), 'test');
-
-  run(function() {
-    view.set('value', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('views within an if statement should be sane on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.display}}{{view view.MyView}}{{/if}}'),
-    MyView: EmberView.extend({
-      tagName: 'input'
-    }),
-    display: false
-  });
-
-  runAppend(view);
-
-  equal(view.$('input').length, 0);
-
-  run(function() {
-    // Setting twice will trigger the observer twice, this is intentional
-    view.set('display', true);
-    view.set('display', 'yes');
-  });
-
-  var textfield = view.$('input');
-  equal(textfield.length, 1);
-
-  // Make sure the view is still registered in View.views
-  ok(EmberView.views[textfield.attr('id')]);
-});
-
-QUnit.test('the {{this}} helper should not fail on removal', function() {
-  view = EmberView.create({
-    context: 'abc',
-    template: compile('{{#if view.show}}{{this}}{{/if}}'),
-    show: true
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'abc', 'should start property - precond');
-
-  run(function() {
-    view.set('show', false);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('properties within an if statement should not fail on re-render', function() {
-  view = EmberView.create({
-    template: compile('{{#if view.value}}{{view.value}}{{/if}}'),
-    value: null
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('value', 'test');
-  });
-
-  equal(view.$().text(), 'test');
-
-  run(function() {
-    view.set('value', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('the {{this}} helper should not fail on removal', function() {
```

Comment:
this test was duplicated, see `the {{this}} helper should not fail on removal` above